### PR TITLE
Jigsaw in `MaterialTag.direction` + Scaffolding in `MaterialTag.type`

### PR DESCRIPTION
**Pull request by tal5**
This pull request adds:

- Support for Jigsaw's "orientation" property in `MaterialTag.direction`
- Support for scaffolding's "bottom" property in `MaterialTag.type`

### CODE REVIEW

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDirectional.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.data.*;
 import org.bukkit.block.data.type.PointedDripstone;
+import org.bukkit.block.data.type.Jigsaw;
+import org.bukkit.block.data.type.Jigsaw.Orientation;
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDirectional.java]: ... wot?

Comment (acikek) [reply plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDirectional.java:18]: presumably leftover from when jigsaw was planned to be in this property

File: plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDirectional.java [hunk 2]
```diff
@@ -300,7 +343,10 @@ public void adjust(Mechanism mechanism) {
             else if (isRail() && mechanism.requireEnum(false, Rail.Shape.values())) {
                 getRail().setShape(Rail.Shape.valueOf(mechanism.getValue().asString().toUpperCase()));
             }
-            else if (mechanism.requireEnum(false, BlockFace.values())) {
+            else if (isJigsaw() && mechanism.requireEnum(false, Orientation.values())) {
+                ((Jigsaw) material.getModernData()).setOrientation(Orientation.valueOf(mechanism.getValue().asString().toUpperCase())); // TODO: 1.16
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialDirectional.java]: this requires a `MultiVersionHelper1_16` akin to dripstone in block type

**Comment by tal5** [comment]
Both fixed.

**Pull request merged**